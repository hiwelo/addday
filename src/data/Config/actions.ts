import { ActionsUnion, ConfigState } from '../types';

/**
 * Resets the current user for this application
 */
function setSentryEnrollment(
  updatedEnrollmentStatus: ConfigState['sentryEnrollment'],
): ActionsUnion {
  return {
    type: 'SENTRY_ENROLLMENT',
    payload: updatedEnrollmentStatus,
  };
}

export { setSentryEnrollment };
