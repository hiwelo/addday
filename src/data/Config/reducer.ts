import { ConfigState, ConfigAction } from './types';

const INITIAL_STATE: ConfigState = {
  sentryEnrollment: false,
};

function configReducer(
  state = INITIAL_STATE,
  action: ConfigAction,
): ConfigState {
  const { type, value } = action;

  switch (type) {
    case 'SENTRY_ENROLLMENT':
      return {
        ...state,
        sentryEnrollment: value.sentryEnrollment,
      };

    default:
      return state;
  }
}

export default configReducer;
