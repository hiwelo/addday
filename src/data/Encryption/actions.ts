import { ActionsUnion } from '../types';

/**
 * Fetches the encryption key
 */
function getEncryptionKey(): ActionsUnion {
  return {
    type: 'GET_ENCRYPTION_KEY',
  };
}

/**
 * Saves a new encryption key
 */
function setEncryptionKey(key: string): ActionsUnion {
  return {
    type: 'SET_ENCRYPTION_KEY',
    payload: key,
  };
}

export { getEncryptionKey, setEncryptionKey };
