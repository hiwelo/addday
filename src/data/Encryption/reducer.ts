import { ActionsUnion, EncryptionState } from '../types';

function encryptionReducer(
  state: EncryptionState = { key: null },
  { type }: ActionsUnion,
): EncryptionState {
  switch (type) {
    case 'GET_ENCRYPTION_KEY':
      return state;

    case 'SET_ENCRYPTION_KEY':
      return {
        ...state,
        key: 'test',
      };

    default:
      return state;
  }
}

export default encryptionReducer;
