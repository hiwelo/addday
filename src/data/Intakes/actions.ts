import { Intake } from '../../models/Intake';
import { ActionsUnion } from '../types';

export function clearIntakes(): ActionsUnion {
  return {
    type: `CLEAR_INTAKES_ACTION`,
  };
}

export function deleteIntake(intakeId: Intake['id']): ActionsUnion {
  return {
    type: `DELETE_INTAKE_ACTION`,
    payload: intakeId,
  };
}

export function setIntake(intake: Intake): ActionsUnion {
  return {
    type: `SET_INTAKE_ACTION`,
    payload: intake,
  };
}
