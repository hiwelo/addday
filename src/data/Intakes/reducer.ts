import { Map } from 'immutable';

import { ActionsUnion, IntakesState } from '../types';

const INITIAL_STATE: IntakesState = Map();

function intakeReducer(
  state: IntakesState = INITIAL_STATE,
  action: ActionsUnion,
): IntakesState {
  switch (action.type) {
    case `CLEAR_INTAKES_ACTION`:
    case `RESET_APP`:
      return state.clear();

    case `DELETE_INTAKE_ACTION`:
      return state.delete(action.payload);

    case `SET_INTAKE_ACTION`:
      return state.set(action.payload.id, action.payload);

    default:
      return state;
  }
}

export default intakeReducer;
