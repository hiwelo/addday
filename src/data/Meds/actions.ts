import { Medication } from '../../models/Medication';
import { ActionsUnion } from '../types';

export function editMedication(medication: Medication): ActionsUnion {
  return {
    type: 'UPDATE_MEDICATION',
    payload: medication,
  };
}
