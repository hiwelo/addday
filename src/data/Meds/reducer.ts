import { Medication, MedsState, MedsAction } from './types';

const INITIAL_STATE: MedsState = {
  medications: {},
};

function medsReducer(state = INITIAL_STATE, action: MedsAction): MedsState {
  const { type, value } = action;

  switch (type) {
    case 'MEDS_ADD':
      return {
        medications: {
          ...state.medications,
          [value.id]: value as Medication,
        },
      };

    default:
      return state;
  }
}

export default medsReducer;
