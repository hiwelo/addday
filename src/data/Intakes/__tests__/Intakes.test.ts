import { mockedIntake } from '../../../models/__mocks__/Intake';
import * as actions from '../actions';
import reducer from '../reducer';

const stateWithExistingIntake = reducer(
  undefined,
  actions.setIntake(mockedIntake),
);

describe('Intakes', () => {
  it('allows you to add a new intake', () => {
    const action = actions.setIntake(mockedIntake);
    const state = reducer(undefined, action);

    expect(action).toMatchSnapshot();
    expect(state).toMatchSnapshot();
  });

  it('allows you to update an existing intake', () => {
    const action = actions.setIntake({
      ...mockedIntake,
      days: {
        ...mockedIntake.days,
        sunday: true,
      },
    });
    const state = reducer(stateWithExistingIntake, action);

    expect(action).toMatchSnapshot();
    expect(state).toMatchSnapshot();
  });

  it('allows you to delete an existing intake', () => {
    const action = actions.deleteIntake(mockedIntake.id);
    const state = reducer(stateWithExistingIntake, action);

    expect(action).toMatchSnapshot();
    expect(state).toMatchSnapshot();
  });

  it('allows you to clear all existing intake', () => {
    const addAction = actions.setIntake({
      ...mockedIntake,
      id: 'test2',
    });
    const updatedState = reducer(stateWithExistingIntake, addAction);
    const action = actions.clearIntakes();
    const state = reducer(updatedState, action);

    expect(action).toMatchSnapshot();
    expect(state).toMatchSnapshot();
  });
});
