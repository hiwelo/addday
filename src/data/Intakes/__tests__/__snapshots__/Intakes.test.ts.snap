// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intakes allows you to add a new intake 1`] = `
Object {
  "payload": Object {
    "days": Object {
      "friday": true,
      "monday": true,
      "saturday": false,
      "sunday": false,
      "thursday": true,
      "tuesday": false,
      "wednesday": false,
    },
    "id": "testScheduledIntake",
    "moment": "2020-01-01T18:00:00.000Z",
  },
  "type": "SET_INTAKE_ACTION",
}
`;

exports[`Intakes allows you to add a new intake 2`] = `
Immutable.Map {
  "testScheduledIntake": Object {
    "days": Object {
      "friday": true,
      "monday": true,
      "saturday": false,
      "sunday": false,
      "thursday": true,
      "tuesday": false,
      "wednesday": false,
    },
    "id": "testScheduledIntake",
    "moment": "2020-01-01T18:00:00.000Z",
  },
}
`;

exports[`Intakes allows you to clear all existing intake 1`] = `
Object {
  "type": "CLEAR_INTAKES_ACTION",
}
`;

exports[`Intakes allows you to clear all existing intake 2`] = `Immutable.Map {}`;

exports[`Intakes allows you to delete an existing intake 1`] = `
Object {
  "payload": "testScheduledIntake",
  "type": "DELETE_INTAKE_ACTION",
}
`;

exports[`Intakes allows you to delete an existing intake 2`] = `Immutable.Map {}`;

exports[`Intakes allows you to update an existing intake 1`] = `
Object {
  "payload": Object {
    "days": Object {
      "friday": true,
      "monday": true,
      "saturday": false,
      "sunday": true,
      "thursday": true,
      "tuesday": false,
      "wednesday": false,
    },
    "id": "testScheduledIntake",
    "moment": "2020-01-01T18:00:00.000Z",
  },
  "type": "SET_INTAKE_ACTION",
}
`;

exports[`Intakes allows you to update an existing intake 2`] = `
Immutable.Map {
  "testScheduledIntake": Object {
    "days": Object {
      "friday": true,
      "monday": true,
      "saturday": false,
      "sunday": true,
      "thursday": true,
      "tuesday": false,
      "wednesday": false,
    },
    "id": "testScheduledIntake",
    "moment": "2020-01-01T18:00:00.000Z",
  },
}
`;
